src/components/HotRepositories.tsx

```diff
@@ -0,0 +1,168 @@
+    name: "Bun",
+    description: "Incredibly fast JavaScript runtime, bundler, transpiler and package manager.",
+    votes: 2,
+    upvoted: false,
+    issues: "503",
+    stars: "27.7k",
+    PR: "262",
+    img: [
+      "https://avatars.githubusercontent.com/u/709451?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/2148168?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/70155278?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/56601352?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/790659?s=64&v=4",
+    ],
+  },
+  {
+    repo_id: 510607652,
+    organization: "Pocketbase",
+    orgImg: "https://avatars.githubusercontent.com/u/101000011?s=200&v=4",
+    name: "Pocketbase",
+    description: "Open Source realtime backend in 1 file",
+    votes: 1,
+    upvoted: false,
+    issues: "72",
+    stars: "7.3k",
+    PR: "32",
+    img: [
+      "https://avatars.githubusercontent.com/u/8248071?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/43366254?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/38179369?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/26606825?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/8593614?s=64&v=4",
+    ],
+  },
+  {
+    repo_id: 71359796,
+    organization: "Open-Sauced",
+    orgImg: "https://avatars.githubusercontent.com/u/57568598?s=200&v=4",
+    name: "Open-Sauced",
+    description: " This is a project to identify your next open source contribution.",
+    votes: 3,
+    upvoted: false,
+    issues: "293",
+    stars: "726",
+    PR: "1k",
+    img: [
+      "https://avatars.githubusercontent.com/u/5713670?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/237133?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/11777161?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/14043845?s=64&v=4",
+      "https://avatars.githubusercontent.com/u/22990146?s=60&v=4",
+    ],
+  },
+];
+
+export declare interface HotReposProps {
+  user: User | null;
+}
+
+const HotRepositories = ({ user }: HotReposProps): JSX.Element => {
+  const {
+    user_metadata: { sub: user_id },
+  } = user || { user_metadata: { sub: null } };
+  const [hotRepos, setHotRepos] = useState(hotRepo);
+
+  // * This function is just a placeholder to help change the color and state of the selected button on the card.
+  const handleVoted = (repo_id: number) => {
+    const votedIdx = hotRepos.findIndex((obj) => obj.repo_id == repo_id);
+    hotRepos[votedIdx].upvoted = !hotRepos[votedIdx].upvoted;
+    setHotRepos([...hotRepos]);
+  };
+
+  async function handleVoteUpdateByRepo(votes: number, repo_id: number) {
+    user_id && capturePostHogAnayltics("User voted", "voteClick", "true");
+
+    await updateVotesByRepo(votes, repo_id, user_id);
+    handleVoted(repo_id);
+  }
+
+  // onClick={() => (user_id ? handleVoteUpdateByRepo(votes, repo_id) : signIn({ provider: "github" }))}
```

Comment:
This is needed to force the user to sign in before voting. Otherwise, we have no idea who submitted the vote. With that said it should be inline in the onClick.

Any chance we can move it down to line 118?

Reply:
This change has been added now. 

🍕